### ISSUE TITLE
Call external solver during SP solve

### ISSUE THREAD

**Issue by bqpd** [hoburg/gpkit#590]
These should be tested, written, used:

For changing a constant:

```python
import numpy as np
from gpkit.constraint.set import ConstraintSet


class ChangeConstantConstraint(ConstraintSet):

    def as_gpconstr(self, x0):
        gpconstr = super(ConstraintSet, self).as_gpconstr(x0)
        if x0["Re"] >= 1e6:
            gpconstr.substitution.update(xfoil(x0))
        return gpconstr


def xfoil(x0):
    return {"CD": np.sin(x0["Re"])}
```

For inserting a constraint:

```python

def xfoil(x0):
    return [PosynomialInequality, MonomialEquality]


class ChangeConstantConstraint(ConstraintSet):

    def as_gpconstr(self, x0):
        gpconstr = super(ConstraintSet, self).as_gpconstr(x0)
        if x0["Re"] >= 1e6:
            constraints = xfoil(x0)
        else:
            constraints = [PosynomialInequality, MonomialEquality]
        return ConstraintSet([gpconstr, constraints])
```

**Comment by bqpd** [comment]
Alternative conditions for calling the outside function, besides variable bounds include
  - sensitivity bounds (upper and lower)

**Comment by cjk12** [comment]
What file should these things live in?

**Comment by bqpd** [comment]
These should live in one of your code files, where they can reference variables and so on. You can wrap all your other constraints in one of these (for the substitutions one) or just the relevant constraints (for the "add a bonus constraint" one).

**Comment by cjk12** [comment]
I just pulled, but I have no module named `constraint`?  Not sure what's going on here.  I also can't find `ConstraintSet` in the source code...

**Comment by cjk12** [comment]
Here's a MWE we can reference:
```python
from gpkit import VectorVariable, Variable, Model, units
import gpkit
from gpkit.constraint import ConstraintSet

CL = Variable("C_L", 1.0, "-", "Lift Coefficient")
CD = Variable("C_D", "-", "Drag Coefficient")

def xfoil(x0):
    return [CD >= 0.01 + .03*CL**2]

class AeroModel(ConstraintSet):

    def as_gpconstr(self, x0):
        gpconstr = super(ConstraintSet, self).as_gpconstr(x0)
        if x0["Re"] >= 1e6:
            constraints = xfoil(x0)
        else:
            constraints = [CD >= 1. + CL**2]
        return ConstraintSet([gpconstr, constraints])
    
objective = CD

m = Model(objective, constraints)
m.solve('cvxopt', verbosity=1)
```

**Comment by bqpd** [comment]
Great example.

**Comment by cjk12** [comment]
Works for me now too.  I'm going to close and raise new issues as they come up.  Thanks @bqpd

**Comment by cjk12** [comment]
nvm, I can't close the issue...  @whoburg

**Comment by whoburg** [comment]
Looking great. Exciting that we can do this so cleanly.

Remaining todos:
 - [ ] should `externalsolver` be turned into a PR?
 - [ ] add example of this to readthedocs

**Comment by whoburg** [comment]
shall we close this now that @cjk12 implemented #687 ?

**Comment by bqpd** [comment]
Looks great! http://gpkit.readthedocs.org/en/latest/signomialprogramming.html#calling-to-external-codes
